pkg/config/common.go

```diff
@@ -0,0 +1,40 @@
+/*
+Copyright 2021 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package config
+
+// Common ExternalName configurations.
+var (
```

Comment:
Sounds good, also we could expose this as a provider-level configuration (e.g. default external name config) that could still be overridden by specific resources. For example, for azure, we can just set `provider.defaultExternalNameConfig = IdentifierFromProvider` and done.

Reply:
Yeah but if we're going to have that for loop anyway, it could be simpler to set it every time so that we'd worry about only a single field in a single config object instead of checking the default if it doesn't exist.

Reply:
Maybe we can find a way not to have that loop :)